examples/vault-auto-unseal/user-data-vault.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# This script is meant to be run in the User Data of each EC2 Instance while it's booting. The script uses the
+# run-consul script to configure and start Consul in client mode and then the run-vault script to configure and start
+# Vault in server mode. Note that this script assumes it's running in an AMI built from the Packer template in
+# examples/vault-consul-ami/vault-consul.json.
+
+set -e
+
+# Send the log output from this script to user-data.log, syslog, and the console
+# From: https://alestic.com/2010/12/ec2-user-data-output/
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+# The Packer template puts the TLS certs in these file paths
+readonly VAULT_TLS_CERT_FILE="/opt/vault/tls/vault.crt.pem"
+readonly VAULT_TLS_KEY_FILE="/opt/vault/tls/vault.key.pem"
+
+# The cluster_tag variables below are filled in via Terraform interpolation
+/opt/consul/bin/run-consul --client --cluster-tag-key "${consul_cluster_tag_key}" --cluster-tag-value "${consul_cluster_tag_value}"
+/opt/vault/bin/run-vault \
+  --tls-cert-file "$VAULT_TLS_CERT_FILE" \
+  --tls-key-file "$VAULT_TLS_KEY_FILE" \
+  --enable-auto-unseal \
+  --auto-unseal-kms-key "${kms_key}" \
+  --auto-unseal-kms-key-region "${aws_region}"
```

Comment:
`--auto-unseal-kms-key` doesn't make it obvious what you're looking for here: ID, ARN, alias, or the actual secret key itself? In the TF code, it looks like you're setting it to ID, so should this be called `--auto-unseal-kms-key-id`?

Reply:
I generally try to avoid the use of "id" in AWS modules. Does ID mean "name" or "ARN"? I never know, and the Terraform docs are inconsistent, so I opt for names like `--auto-unseal-kms-key-arn` or `--auto-unseal-kms-key-alias`

Reply:
It means the `key_id` output https://www.terraform.io/docs/providers/aws/r/kms_key.html

Reply:
Yea, ID is the proper term here. It has a concrete definition with KMS CMKs. The ARN and name (actually, alias) are separate for a CMK. We want ID here.

Reply:
It is neither the name, neither the arn. It is the key itself, which everything else refers to key id, so I'm specifying id to make it clearer.